apis/python/src/tiledbsoma/_geometry_dataframe.py

```diff
@@ -90,7 +120,163 @@ def create(
             Experimental.
         """
         warnings.warn(SPATIAL_DISCLAIMER)
-        raise NotImplementedError()
+        if not isinstance(coordinate_space, CoordinateSpace):
+            coordinate_space = CoordinateSpace.from_axis_names(coordinate_space)
```

Comment:
Is this check robust enough? Typeguard will preemptively throw a type error for anything that isn't `Sequence[str]` or `CoordinteSpace` before it gets here?